newIDE/app/src/UI/Menu/MaterialUIMenuImplementation.js

```diff
@@ -215,13 +218,17 @@ export default class MaterialUIMenuImplementation {
                 // $FlowFixMe - existence should be inferred by Flow.
                 item.enabled === false
               }
-              onClick={() => {
+              onClick={async () => {
                 if (item.enabled === false) {
                   return;
                 }
 
                 if (item.click) {
-                  item.click();
+                  await item.click();
+                  if (item.type === 'checkbox') {
+                    forceUpdate();
```

Comment:
Would really benefit from a comment explaining how the heck this helps? 😅

Reply:
Ahah yes of course